src/freenet/support/io/FileUtil.java

```diff
@@ -722,11 +722,9 @@ public static boolean copyFile(File copyFrom, File copyTo) {
 	 * @throws IOException If unable to write to the stream.
 	 */
 	public static void fill(OutputStream os, long length) throws IOException {
-		long seed;
-		synchronized (SEED_GENERATOR) {
-			seed = SEED_GENERATOR.nextLong();
-		}
-		writeRandomBytes(os, new MersenneTwister(seed), length);
+		byte[] seed = new byte[16];
+		SEED_GENERATOR.nextBytes(seed);
+		writeRandomBytes(os, MersenneTwister.createUnsynchronized(seed), length);
```

Comment:
This uses a longer seed to generate the new twister, but I’m guessing that’s both irrelevant and an improvement (if you assume longer seeds are “better”), right? 😁